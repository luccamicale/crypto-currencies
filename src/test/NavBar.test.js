import { render, screen } from '@testing-library/react';
import { StrictMode } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import NavBar from '../components/Navbar';
import store from '../redux/configureStore';

describe('Testing Navbar page', () => {
  it('Should match the navbar snap', () => {
    const companies = render(
      <StrictMode>
        <Router>
          <Provider store={store}>
            <NavBar />
          </Provider>
        </Router>
      </StrictMode>,
    );
    expect(companies).toMatchSnapshot();
  });
  it('Should have text Crypto', () => {
    render(
      <Provider store={store}>
        <Router>
          <NavBar />
        </Router>
      </Provider>,
    );
    expect(screen.getByText('Crypto')).toBeInTheDocument();
  });
});
