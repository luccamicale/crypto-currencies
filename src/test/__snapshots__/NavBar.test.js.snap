// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navbar page Should match the navbar snap 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nav-container"
      >
        <div
          class="sub-container"
        >
          <div
            class="container-bitcoin"
          >
            <img
              alt="img-bitcoin"
              class="bitcoin"
              src="bitcoin.png"
            />
          </div>
          <div
            class="container-link"
          >
            <a
              aria-current="page"
              class="title active"
              href="/"
            >
              Crypto
            </a>
          </div>
          <div
            class="container-casa"
          >
            <img
              alt="img-casa"
              class="carro"
              src="carro.png"
            />
          </div>
          <div
            class="container-carro"
          >
            <img
              alt="img-casa"
              class="casa"
              src="casa.png"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nav-container"
    >
      <div
        class="sub-container"
      >
        <div
          class="container-bitcoin"
        >
          <img
            alt="img-bitcoin"
            class="bitcoin"
            src="bitcoin.png"
          />
        </div>
        <div
          class="container-link"
        >
          <a
            aria-current="page"
            class="title active"
            href="/"
          >
            Crypto
          </a>
        </div>
        <div
          class="container-casa"
        >
          <img
            alt="img-casa"
            class="carro"
            src="carro.png"
          />
        </div>
        <div
          class="container-carro"
        >
          <img
            alt="img-casa"
            class="casa"
            src="casa.png"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
